src/main/java/com/artipie/asto/s3/ContentOfFuture.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2020 artipie.com
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.artipie.asto.s3;
+
+import com.artipie.asto.Content;
+import java.nio.ByteBuffer;
+import java.util.Optional;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.ExecutionException;
+import org.reactivestreams.Subscriber;
+
+/**
+ * Content of a future.
+ *
+ * @since 1.0
+ */
+final class ContentOfFuture implements Content {
+    /**
+     * The future.
+     */
+    private final CompletionStage<Content> original;
+
+    /**
+     * Ctor.
+     *
+     * @param future The future to provide content.
+     */
+    ContentOfFuture(final CompletionStage<Content> future) {
+        this.original = future;
+    }
+
+    @Override
+    public Optional<Long> size() {
+        return this.complete().size();
+    }
+
+    @Override
+    public void subscribe(final Subscriber<? super ByteBuffer> subscriber) {
+        this.complete().subscribe(subscriber);
+    }
+
+    /**
+     * Complete the future and handle checked exceptions.
+     *
+     * @return The content.
+     */
+    private Content complete() {
+        try {
+            return this.original.toCompletableFuture().get();
+        } catch (final InterruptedException ex) {
```

Comment:
@genryxy It would, but it's returned from another `Future`.

Reply:
@andreoss please redesign this class in some way to avoid using `get()` method in not test code

Reply:
@genryxy Any suggestions how? Can I leave a todo in order to continue working on it?
I don't see an issue here, because
-  `Content` seems to have a blocking API anyway 
- It is a part of a Future which is Returned from `Storage`

Reply:
@andreoss `Content` does not have a blocking API.  
I think this class may not implement `Content` interface and one of the method of this class could return `CompletionStage<Content>` because now you wrap estimated content into future:
```
result = CompletableFuture
                .completedFuture(new EstimatedContent(onetime))
                .thenCompose(updated -> this.put(key, updated));
```

Reply:
@genryxy Done.